tree.go

```diff
@@ -823,11 +833,11 @@ func (n *InternalNode) Commit() *Point {
 		minBatchSize := 4
 		if len(nodes) <= minBatchSize {
 			if err := commitNodesAtLevel(nodes); err != nil {
-				// TODO: make Commit() return an error
-				panic(err)
+				return nil, fmt.Errorf("committing nodes at level %d: %w", level, err)
 			}
 		} else {
 			var wg sync.WaitGroup
+			errChan := make(chan error, 1)
```

Comment:
this is incorrect, as more than one goroutine can write an error to the channel. If so, multiple goroutines can leak.